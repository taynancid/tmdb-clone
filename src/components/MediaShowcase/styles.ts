import styled from 'styled-components'

export const Container = styled.div``
export const Header = styled.div``
export const Title = styled.span``
export const TabsWrapper = styled.div``
export const Body = styled.div``
