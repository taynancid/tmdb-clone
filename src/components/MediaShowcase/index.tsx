import * as S from './styles'

export type Tab = {
  title: string
  url: string
}

export type Props = {
  title: string
  tabs: Tab[]
}

const MediaShowcase = ({ title, tabs }: Props) => {
  return (
    <S.Container>
      <S.Header>
        <S.Title>{title}</S.Title>
        <S.TabsWrapper>{tabs.map((tab) => tab.title)}</S.TabsWrapper>
      </S.Header>
      <S.Body>movies</S.Body>
    </S.Container>
  )
}

export default MediaShowcase
