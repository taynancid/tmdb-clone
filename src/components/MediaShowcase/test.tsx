import { render, screen } from '@testing-library/react'

import MediaShowcase from '.'
import { Tab } from './index'

describe('<MediaShowcase />', () => {
  it('should render the title', () => {
    const title = 'Trending'
    const tabs: Tab[] = [
      { title: 'movies', url: '/api/movies' },
      { title: 'tv shows', url: '/api/tv-shows' }
    ]
    const { container } = render(<MediaShowcase title={title} tabs={tabs} />)

    expect(screen.getByText(/trending/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
