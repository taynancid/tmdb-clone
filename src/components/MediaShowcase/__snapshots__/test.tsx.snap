// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaShowcase /> should render the title 1`] = `
<div
  class=""
>
  <div
    class=""
  >
    <span
      class=""
    >
      Trending
    </span>
    <div
      class=""
    >
      movies
      tv shows
    </div>
  </div>
  <div
    class=""
  >
    movies
  </div>
</div>
`
